components/brave_wallet/browser/solana_response_parser.cc

```diff
@@ -77,28 +104,86 @@ bool ParseGetTokenAccountBalance(const base::Value& json_value,
     return false;
   }
 
-  auto* amount_ptr = value->FindString("amount");
-  if (!amount_ptr) {
+  auto parsed_amount = ParseAmountDict(*value);
+  if (!parsed_amount) {
     return false;
   }
-  *amount = *amount_ptr;
 
-  // uint8
-  auto decimals_opt = value->FindInt("decimals");
-  if (!decimals_opt ||
-      decimals_opt.value() > std::numeric_limits<uint8_t>::max() ||
-      decimals_opt.value() < 0) {
-    return false;
+  *amount = std::get<0>(*parsed_amount);
+  *ui_amount_string = std::get<1>(*parsed_amount);
+  *decimals = std::get<2>(*parsed_amount);
+  return true;
+}
+
+absl::optional<std::vector<std::tuple<std::string,  // mint
+                                      std::string,  // amount
+                                      std::string,  // ui amount
+                                      uint8_t>>>    // decimals
+ParseGetSPLTokenBalances(const base::Value& json_value) {
+  auto result = ParseResultDict(json_value);
+  if (!result) {
+    return absl::nullopt;
   }
-  *decimals = decimals_opt.value();
 
-  auto* ui_amount_string_ptr = value->FindString("uiAmountString");
-  if (!ui_amount_string_ptr) {
-    return false;
+  auto* value = result->FindList("value");
+  if (!value) {
+    return absl::nullopt;
   }
-  *ui_amount_string = *ui_amount_string_ptr;
 
-  return true;
+  std::vector<std::tuple<std::string, std::string, std::string, uint8_t>>
+      balances;
+
+  for (const auto& account_value : *value) {
+    if (!account_value.is_dict()) {
+      return absl::nullopt;
+    }
+
+    const auto& account_dict = account_value.GetDict();
+    auto* account = account_dict.FindDict("account");
+    if (!account) {
+      return absl::nullopt;
+    }
+
+    auto* data = account->FindDict("data");
+    if (!data) {
+      return absl::nullopt;
+    }
+
+    auto* parsed = data->FindDict("parsed");
+    if (!parsed) {
+      return absl::nullopt;
+    }
+
+    auto* info = parsed->FindDict("info");
+    if (!info) {
+      return absl::nullopt;
+    }
+
+    auto* mint = info->FindString("mint");
+    if (!mint) {
+      return absl::nullopt;
+    }
+
+    auto* tokenAmount = info->FindDict("tokenAmount");
+    if (!tokenAmount) {
+      return absl::nullopt;
+    }
+
+    auto parsed_amount = ParseAmountDict(*tokenAmount);
+    if (!parsed_amount) {
+      return absl::nullopt;
+    }
```

Comment:
Would be shorted if written as:
FindStringByDottedPath("account.data.parsed.info.mint")
FindDictByDottedPath("account.data.parsed.info.tokenAmount")

Reply:
f4a31ca